zrml/swaps/src/tests.rs

```diff
@@ -3015,6 +3019,225 @@ fn pool_exit_with_exact_asset_amount_fails_if_liquidity_drops_too_low() {
+        assert_eq!(Currencies::free_balance(ASSET_D, &pool_account_id), balance + arbitrage_amount);
+        assert_eq!(
+            Currencies::free_balance(ASSET_B, &pool_account_id),
+            balance + arbitrage_amount - amount_removed,
+        );
+        let market_id = 0;
+        let market_account_id = MarketCommons::market_account(market_id);
+        assert_eq!(Currencies::free_balance(ASSET_A, &market_account_id), arbitrage_amount);
+        System::assert_has_event(Event::ArbitrageMintSell(pool_id, arbitrage_amount).into());
+    });
+}
+
+#[test]
+fn on_idle_arbitrages_pools_with_buy_burn() {
+    ExtBuilder::default().build().execute_with(|| {
+        frame_system::Pallet::<Runtime>::set_block_number(1);
+        let assets = ASSETS;
+        assets.iter().cloned().for_each(|asset| {
+            assert_ok!(Currencies::deposit(asset, &BOB, _10000));
+        });
+        let balance = <Runtime as crate::Config>::MinLiquidity::get();
+        assert_ok!(Swaps::create_pool(
+            BOB,
+            assets.into(),
+            ASSET_A,
+            0,
+            ScoringRule::CPMM,
+            Some(0),
+            Some(balance),
+            Some(vec![_3, _1, _1, _1]),
+        ));
+        let pool_id = 0;
+
+        // Withdraw a certain amount of base tokens to push the total spot price below 1 (ASSET_A
+        // is the base asset, all other assets are considered outcomes).
+        let pool_account_id = Swaps::pool_account_id(&pool_id);
+        let amount_removed = _25;
+        assert_ok!(Currencies::withdraw(ASSET_A, &pool_account_id, amount_removed));
+
+        // Deposit funds into the prize pool to ensure that the transfers don't fail.
+        let market_id = 0;
+        let market_account_id = MarketCommons::market_account(market_id);
+        let arbitrage_amount = 125_007_629_394; // "Should" be 125_000_000_000.
+        assert_ok!(Currencies::deposit(
+            ASSET_A,
+            &market_account_id,
+            arbitrage_amount + SENTINEL_AMOUNT,
+        ));
+
+        // Force arbitrage hook.
+        crate::PoolsCachedForArbitrage::<Runtime>::insert(pool_id, ());
+        Swaps::on_idle(System::block_number(), Weight::max_value());
+        // assert_ok!(Swaps::execute_arbitrage(pool_id));
+
+        assert_eq!(
+            Currencies::free_balance(ASSET_A, &pool_account_id),
+            balance + arbitrage_amount - amount_removed,
+        );
+        assert_eq!(Currencies::free_balance(ASSET_B, &pool_account_id), balance - arbitrage_amount);
+        assert_eq!(Currencies::free_balance(ASSET_C, &pool_account_id), balance - arbitrage_amount);
+        assert_eq!(Currencies::free_balance(ASSET_D, &pool_account_id), balance - arbitrage_amount);
+        assert_eq!(Currencies::free_balance(ASSET_A, &market_account_id), SENTINEL_AMOUNT);
+        System::assert_has_event(Event::ArbitrageBuyBurn(pool_id, arbitrage_amount).into());
+    });
+}
```

Comment:
Maybe it could be useful to use a variable for ASSET_A as `base_asset`.

Reply:
be1411bf7aa9468d780de27d5393b6bad54c9c79